client.go

```diff
@@ -638,6 +622,30 @@ func (c *Client) close(handle string) error {
 	}
 }
 
+func (c *Client) stat(path string) (*FileStat, error) {
```

Comment:
It’s kind of odd that we were converting the `FileStat` from `Stat` into an `os.FileInfo` to extract the size, while the fstat didn’t. This aligns the two to work the same way, which simplifies the use point in `WriteTo`.